src/spikeinterface/core/base.py

```diff
@@ -56,7 +56,7 @@ def __init__(self, main_ids: Sequence) -> None:
 
         # "main_ids" will either be channel_ids or units_ids
         # They are used for properties
-        self._main_ids = np.array(main_ids)
+        self._main_ids = np.asarray(main_ids)
```

Comment:
This is cool, I did not know this difference! (`array` copies, `asarray` doesn't). In this context, would it be safer to make a copy? I am not very familiar with this part of the codebase, it possible that someone passes an array of ids here from their workspace and it might be silently mutated?

Reply:
Or actually, this is completely internal read from SI metadata in saved files?

Reply:
Usually is an internal thing, those come directly with the format and not with the user interface. But maybe it is safer to make a copy just in case someone builds an extractor or modifies the ids somewhere else.

Not sure.

Reply:
Will keep it as it was meanwhile.

Reply:
Reverted so I will mark this as resolved.